Tasks/DotNetCoreInstallerV1/Task Design.md

```diff
@@ -0,0 +1,110 @@
+     - `installationPath` should contains a folder called `sdk or host/fxr`. If it
+       - `exists:` check if sdk or host/fxr folder contains a folder with name = version to be downloaded.
+         - `exists:` Version is cached, nothing more needs to be done.
+         - `doesn't exist:` Version needs to be installed.
+       - `doesn't exist:` Version needs to be installed
+
+2. **`version.completed file based approach`**
+   - `Installation process:` Once all the files are successfully extracted, copied and pasted into installationPath, we will create a **`version.completed`** file in sdk/runtime folder. The version.completed file will exist at the same level as the version folder and will be considered as a stamp of successfull installation.
+     - If installation is not complete/successfull, we will not create the version.completed file.
+   - `Caching Process:` So, the algorithm will be same as in Approach 1, but with an additional required check for `version.completed` file to be present in sdk or host/fxr folder along with folder with name = version. Based on version.completed's existance, the following will be concluded
+     - `Exists`: version will be considered as cached
+     - `Doesn't exist`: version will be installed. (overwriting if the files already exist)
+     - **optionally:** `Doesn't exist but installationPath is global .Net installation path`: in this case the version will be considered as cached even without the version.completed file.
+
+
+There are a few advantages of `version.completed` based approach over `folder only` based approach:
+- Handles failed installation attempts: in case the last installation was left incomplete, the version.completed file would not exist and thus in next try, the version will be installed.
+
+
+## **- Inbuilt option to restrict Multi-Level lookup**
+This feature id introduced as users had asked for an option to disable Multi-Level lookup, to know more about user's ask, refer [here](https://github.com/Microsoft/azure-pipelines-tasks/issues/9608).
+
+Multi-level lookup configures the .Net core host to probe several locations to find a suitable shared framework. If a folder is not found in folder containing dotnet.exe, it will attempt to look in pre-defined global locations using multi-level lookup. The default global locations are:
+
+**For Windows:**
+- C:/Program Files/dotnet (64-bit processes)
+- C:/Program Files (x86)/dotnet
+
+
+**For MacOs:**
+- /usr/local/share/dotnet
+
+
+**For Linux:**
+- /usr/share/dotnet
+
+It's value will be set as an environment variable: `DOTNET_MULTILEVEL_LOOKUP` with values
+- `0` : Disabled
+- `1` : Enabled
+
+For more information on Multi-Level Lookup refer [HERE](https://github.com/dotnet/core-setup/blob/master/Documentation/design-docs/multilevel-sharedfx-lookup.md).
```

Comment:
Will you also add details for global.json support?

Reply:
The support should be added, however not thinking of doing it immediately.